src/txn/txn.c

```diff
@@ -381,7 +396,8 @@ __wt_txn_release(WT_SESSION_IMPL *session)
 	 */
 	__wt_txn_release_snapshot(session);
 	txn->isolation = session->isolation;
-	F_CLR(txn, WT_TXN_ERROR | WT_TXN_HAS_ID | WT_TXN_RUNNING);
+	F_CLR(txn, WT_TXN_ERROR | WT_TXN_HAS_ID |
+	    WT_TXN_NAMED_SNAPSHOT | WT_TXN_READONLY | WT_TXN_RUNNING);
```

Comment:
The ```TXN_API_END_RETRY``` macro is what I'm concerned about. It uses a combination of ```__autotxn``` (i.e: the flag was set on ```TXN_API_CALL```. It assumes that if WT_TXN_AUTOCOMMIT has been cleared, then a transaction was created for the operation that completed, and needs to be committed.

Could be that I'm missing something.